worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala

```diff
@@ -526,4 +530,278 @@ class PushDataHandler extends BaseMessageHandler with Logging {
+        messageType match {
+          case Type.PUSH_MERGED_DATA => callback.onFailure(new Exception(msg))
+          case _ => callback.onFailure(
+              new Exception(StatusCode.PUSH_DATA_FAIL_PARTITION_NOT_FOUND.getMessage()))
+        }
+      }
+      return true
+    }
+    false
+  }
+
+  private def checkFileWriterException(
+      messageType: Message.Type,
+      isMaster: Boolean,
+      fileWriter: FileWriter,
+      callback: RpcResponseCallback): Unit = {
+    logWarning(
+      s"[handle$messageType] fileWriter $fileWriter has Exception ${fileWriter.getException}")
+
+    val (messageMaster, messageSlave) =
+      messageType match {
+        case Type.PUSH_DATA | Type.PUSH_DATA_HAND_SHAKE =>
+          (
+            StatusCode.PUSH_DATA_FAIL_MASTER.getMessage(),
+            StatusCode.PUSH_DATA_FAIL_SLAVE.getMessage())
+        case Type.PUSH_DATA_HAND_SHAKE => (
+            StatusCode.PUSH_DATA_HANDSHAKE_FAIL_MASTER.getMessage,
+            StatusCode.PUSH_DATA_HANDSHAKE_FAIL_SLAVE.getMessage)
+        case Type.REGION_START => (
+            StatusCode.REGION_START_FAIL_MASTER.getMessage,
+            StatusCode.REGION_START_FAIL_SLAVE.getMessage)
+        case Type.REGION_FINISH => (
+            StatusCode.REGION_FINISH_FAIL_MASTER.getMessage,
+            StatusCode.REGION_FINISH_FAIL_SLAVE.getMessage)
+      }
+    callback.onFailure(new Exception(
+      if (isMaster) messageMaster else messageSlave,
+      fileWriter.getException))
+  }
+
+  private def getFileWriterAndCheck(
+      messageType: Message.Type,
+      location: PartitionLocation,
+      isMaster: Boolean,
+      callback: RpcResponseCallback): (Boolean, FileWriter) = {
+    val fileWriter = location.asInstanceOf[WorkingPartition].getFileWriter
+    val exception = fileWriter.getException
+    if (exception != null) {
+      checkFileWriterException(messageType, isMaster, fileWriter, callback)
+      return (true, fileWriter)
+    }
+    (false, fileWriter)
+  }
+
+  private def checkDiskFull(fileWriter: FileWriter): Boolean = {
+    val diskFull = workerInfo.diskInfos
+      .get(fileWriter.flusher.asInstanceOf[LocalFlusher].mountPoint)
+      .actualUsableSpace < diskReserveSize
+    diskFull
+  }
+
+  private def checkDiskFullAndSplit(
+      fileWriter: FileWriter,
+      isMaster: Boolean,
+      softSplit: AtomicBoolean,
+      callback: RpcResponseCallback): Boolean = {
+    val diskFull = checkDiskFull(fileWriter)
+    if ((diskFull && fileWriter.getFileInfo.getFileLength > partitionSplitMinimumSize) ||
+      (isMaster && fileWriter.getFileInfo.getFileLength > fileWriter.getSplitThreshold())) {
+      if (softSplit != null && fileWriter.getSplitMode == PartitionSplitMode.SOFT) {
+        softSplit.set(true)
+      } else {
+        callback.onSuccess(ByteBuffer.wrap(Array[Byte](StatusCode.HARD_SPLIT.getValue)))
+        return true
+      }
+    }
+    false
+  }
+
+  private def getMapAttempt(
+      body: ByteBuf,
```

Comment:
The body variable is always null at present. Will there add a refactor pull request to merge map partition and reduce partition push logic?

Reply:
yes , refatcor will do after this pr.